trendico/views.py

```diff
@@ -4,19 +4,53 @@
 from django.views import View
 from django.core.paginator import Paginator
 from django.db.models import Sum
-from trendico.models import Product
+from trendico.models import Product,ProductCategory
 from trendico.forms import SignUpForm
-
-
+from django.http import JsonResponse
+import json
 # Create your views here.
 
 class HomeView(View):
-    def get(self,request):
-        products=Product.objects.all()
-        products = Product.objects.annotate(
-            remaining_quantity=Sum('stock__quantity')
+    def get(self, request, category=None):
+        categories = ProductCategory.objects.all()
+        products = Product.objects.all()
+
+        if category:
+            products = Product.get_specific_products(category)
+
+        products = products.annotate(
+            remaining_quantity=Sum('stocks__quantity')
         )
-        return render(request,'index.html',{'Products':products})
+        context = {
+            'categories': categories,
+            'Products': products,
+            'selected_category': category
+        }
+        return render(request, 'index.html', context)
+
+class ProductsByCategoryView(View):
+    def get(self, request, category):
+        products = Product.get_specific_products(category)
+
+        products = products.annotate(
+            remaining_quantity=Sum('stocks__quantity')
+        )
+
+        products_data = []
+        for product in products:
+            product_data = {
+                'name': product.name,
+                'price': product.price,
+                'discount_price': product.discount_price,
+                'category': product.category.name,
+                'description': product.description,
+                'image_url': product.images.first().image.url if product.images.first() else None,
```

Comment:
shouldn't we send `product.images.all()`
this will fail in case there are multiple images for a product, but we show only one of them

Reply:
as in new products category section we just want 1 picture to show but when a detailed page will be opened for the same product there we want all images is that so or i am taking this in wrong way?

Reply:
this case seems fine, i though we are showing this on home page